pkg/cloudprovider/providers/gce/cloud/gen.go

```diff
@@ -0,0 +1,10446 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file was generated by "go run gen/main.go > gen.go". Do not edit
+// directly.
+
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/golang/glog"
+	"google.golang.org/api/googleapi"
+
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/filter"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/meta"
+
+	alpha "google.golang.org/api/compute/v0.alpha"
+	beta "google.golang.org/api/compute/v0.beta"
+	ga "google.golang.org/api/compute/v1"
+)
+
+// Cloud is an interface for the GCE compute API.
+type Cloud interface {
+	Addresses() Addresses
+	AlphaAddresses() AlphaAddresses
+	BetaAddresses() BetaAddresses
+	GlobalAddresses() GlobalAddresses
+	BackendServices() BackendServices
+	AlphaBackendServices() AlphaBackendServices
+	AlphaRegionBackendServices() AlphaRegionBackendServices
+	Disks() Disks
+	AlphaDisks() AlphaDisks
+	AlphaRegionDisks() AlphaRegionDisks
+	Firewalls() Firewalls
+	ForwardingRules() ForwardingRules
```

Comment:
Where do we maintain the mapping/docs from GCE concepts (ForwardingRules) or K8S concepts (LoadBalancer)?

Reply:
That is a TODO that needs to be done (but probably can't be addressed here...)